parquet-encoding/src/main/java/org/apache/parquet/bytes/BytesInput.java

```diff
@@ -247,7 +248,10 @@ public void writeAllTo(OutputStream out) throws IOException {
 
     public byte[] toByteArray() throws IOException {
       if (DEBUG) LOG.debug("read all "+ byteCount + " bytes");
-      byte[] buf = new byte[byteCount];
+      if (buf != null) {
+        return buf;
+      }
+      this.buf = new byte[byteCount];
```

Comment:
I'd rather we treat BytesInput as not reusable.
See writeAllTo() above line 243.
This is one of the remaining copies that we should get rid of.
This change forces to copy the content on heap.